core/src/test/scala/akka/persistence/postgres/journal/PostgresJournalSpec.scala

```diff
@@ -54,6 +55,119 @@ abstract class PostgresJournalSpec(config: String, schemaType: SchemaType)
+    val msg = AtomicWrite(
+      PersistentRepr(
+        payload = s"a-$seqNr",
+        sequenceNr = seqNr,
+        persistenceId = pid,
+        sender = sender,
+        writerUuid = writerUuid))
+    val probe = TestProbe()
+    journal ! WriteMessages(List(msg), probe.ref, actorInstanceId)
+    probe.expectMsg(WriteMessagesSuccessful)
+  }
+
+  "A journal" must {
+    "not allow to store events with sequence number lower than what is already stored for the same persistence id" in {
+      // given
+      val perId = "perId"
+      val sender = TestProbe()
+      val repeatedSnr = 5
+
+      // when
+      writeMessages(1, repeatedSnr + 1, perId, sender.ref, writerUuid)
+
+      // then
+      val msg = AtomicWrite(
+        PersistentRepr(
+          payload = s"a-$repeatedSnr",
+          sequenceNr = repeatedSnr,
+          persistenceId = perId,
+          sender = sender.ref,
+          writerUuid = writerUuid))
+
+      val probe = TestProbe()
+      journal ! WriteMessages(Seq(msg), probe.ref, actorInstanceId)
+      probe.expectMsgType[WriteMessagesFailed]
+    }
+  }
+
+  "An insert on the journal" must {
+    import akka.persistence.postgres.db.ExtendedPostgresProfile.api._
+
+    val metadataTable = JournalMetadataTable(journalConfig.journalMetadataTableConfiguration)
+
+    "automatically insert journal metadata" in {
+      // given
+      val perId = "perId-meta-1"
+      val sender = TestProbe()
+
+      // when
+      writeSingleMessage(1, perId, sender.ref, writerUuid)
+
+      // then
+      val metadataExists = db.run(metadataTable.filter(_.persistenceId === perId).exists.result).futureValue
+      metadataExists shouldBe true
+    }
+
+    "upsert only max_sequence_number and max_ordering if metadata already exists" in {
+      // given
+      val perId = "perId-meta-2"
+      val sender = TestProbe()
+      writeSingleMessage(1, perId, sender.ref, writerUuid)
+      val prevMaxSeqNr =
+        db.run(metadataTable.filter(_.persistenceId === perId).map(_.maxSequenceNumber).result.head).futureValue
+      val prevMaxOrdering =
+        db.run(metadataTable.filter(_.persistenceId === perId).map(_.maxOrdering).result.head).futureValue
+      val prevMinOrdering =
+        db.run(metadataTable.filter(_.persistenceId === perId).map(_.minOrdering).result.head).futureValue
+
+      // when
+      writeSingleMessage(2, perId, sender.ref, writerUuid)
+
+      // then
+      val newMaxSeqNr =
+        db.run(metadataTable.filter(_.persistenceId === perId).map(_.maxSequenceNumber).result.head).futureValue
+      val newMaxOrdering =
+        db.run(metadataTable.filter(_.persistenceId === perId).map(_.maxOrdering).result.head).futureValue
+      val newMinOrdering =
+        db.run(metadataTable.filter(_.persistenceId === perId).map(_.minOrdering).result.head).futureValue
+
+      newMaxSeqNr shouldBe prevMaxSeqNr + 1
+      newMaxOrdering shouldBe prevMaxOrdering + 1
+      newMinOrdering shouldBe prevMinOrdering
```

Comment:
suggestion: Add a test that the `newMinOrdering/prevMinOrdering` is _not_ -1 or 0 (greater than 0)

Reply:
Added this check 👍